src/secp256k1.c

```diff
@@ -83,29 +83,43 @@ size_t secp256k1_context_prealloc_size(unsigned int flags) {
     return ret;
 }
 
-secp256k1_context* secp256k1_context_create(unsigned int flags) {
-    secp256k1_context* ret = (secp256k1_context*)checked_malloc(&default_error_callback, sizeof(secp256k1_context));
+secp256k1_context* secp256k1_context_prealloc_create(void* prealloc, unsigned int flags) {
+    void* const base = prealloc;
+    size_t const ctx_size = secp256k1_context_prealloc_size(flags);
+    secp256k1_context* ret = (secp256k1_context*)manual_alloc(&prealloc, sizeof(secp256k1_context), base, ctx_size);
+
     ret->illegal_callback = default_illegal_callback;
     ret->error_callback = default_error_callback;
 
     if (EXPECT((flags & SECP256K1_FLAGS_TYPE_MASK) != SECP256K1_FLAGS_TYPE_CONTEXT, 0)) {
             secp256k1_callback_call(&ret->illegal_callback,
                                     "Invalid flags");
-            free(ret);
+            /* Unreachable */
```

Comment:
This isn't unreachable, is it? If the user provides bad flags it will trigger.

Reply:
We call `default_illegal_callback` here, which aborts, so the return is unreachable.

(By the way I think the diff is somewhat misleading. I'm not adding `/* Unreachable */` because I'm removing the `free`; the two changes are unrelated.)

Reply:
Oh, yeah, you're right.